src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/CastParser.java

```diff
@@ -63,15 +63,14 @@ public String parse(boolean isReturn) throws SqlParseException {
             } else if (DataType.valueOf(dataType) == DataType.STRING) {
                 result.add(String.format("def %s = %s.toString()", name, fileName));
             } else if (DataType.valueOf(dataType) == DataType.DATETIME) {
-                result.add(String.format("def %s = new Date(Double.parseDouble(%s.toString()).longValue())",
-                        name, fileName));
+                result.add(String.format("def %s = new SimpleDateFormat(\"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'\")"
```

Comment:
So the assumption is the field should be in this format. Not sure if it's easy to implement but MySQL does this in  more intelligent way.

```
>create table test (day text);
>
>insert into test values ('2019-11-06');
>insert into test values ('2019');
>insert into test values ('2019-11-06 12');
>
>select cast(day as datetime) from test;
cast(day as datetime)
2019-11-06 00:00:00
NULL
2019-11-06 12:00:00
```

Reply:
It will convert other date formats into that format as well- I tested an example with "2019-03-02" and it converted it into that SimpleDateFormat

Reply:
> It will convert other date formats into that format as well- I tested an example with "2019-03-02" and it converted it into that SimpleDateFormat

Cool, thanks! Maybe you can add test cases for field with different format (after you address my first comment by creating a keyword field)